src/libraries/System.Runtime/tests/System/ArrayTests.cs

```diff
@@ -1853,6 +1863,34 @@ public static void CreateInstance_Advanced(Type elementType, int[] lengths, int[
             Array array6 = Array.CreateInstance(elementType, lengths, lowerBounds);
             VerifyArray(array6, elementType, lengths, lowerBounds, repeatedValue);
         }
+        [Fact]
+        public void CreateInstance_InvokeCanon_Success()
```

Comment:
I'm missing what is being validated here by using `System.__Canon`.

Reply:
Ah just testing more code paths covered by Array.CreateInstance/Type.MakeTypeArray. Since this PR initially started out as tests for Array.CreateInstance in rather odd scenarios, I''m happy keeping it in this PR

Reply:
What's the value of testing this though? This is hardly something I would consider a supported scenario. The fact that the `CreateInstance` doesn't fail could also be seen as a bug. I actually see `Type.GetType("System.__Canon");` not returning a null a bug.

Mono has similar internal types used to implement generic sharing (`Mono.ValueTuple` and friends). Do we need to test those too?

Reply:
The only way one can get `__Canon` is via private reflection. We make no guarantees about the runtime behavior when somebody gets the internal implementation details via private reflection. These tests should be deleted. If there are paths in the runtime that are only reachable via these tests, I would be fine with deleting those too.

Reply:
@hughbe To clarify if you aren't aware of the concept: `System.__Canon` is an implementation detail of the runtime, this type is used as placeholder when something is shared between multiple generic instantiations (e.g. something is valid for both `List<ClassA>` and `List<ClassB>` you'd see `List<__Canon>` as the type its applicable to, e.g. in a profiler).

If someone creates a _concrete type_ with `__Canon` as element or generic parameter this is running into undefined behavior territory, as it can no longer be distinguished from the placeholder. You'd at the very least break profilers, if not the runtime itself.

As such testing creating type instantiations with `__Canon` sounds bogus.